azurelinuxagent/common/future.py

```diff
@@ -130,4 +130,15 @@ def get_openwrt_platform():
             elif product_matches:
                 if product_matches.group(1) == "OpenWrt":
                     result[0] = "openwrt"
-    return result
\ No newline at end of file
+    return result
+
+
+def array_to_string_or_bytes(buff):
```

Comment:
NIT: Wouldn't this make more sense in the `textutil` module? 
Basically have a check for like `is_py_39` in `future.py` and then this function in `textutil`?

A minor suggestion but I'm good with this too :)

Reply:
The way I saw it was that this is a breaking change for future Py versions so it would make sense to have it in `future.py`. That module already contains other breaking changes like the move from `platform` to `distro`. But now that you mention `textutil`, I see we have some version-specific code there too, so that also makes sense.

Anyways, if somebody else feels strongly about this I can make the change, but I don't mind it as it is now.

Reply:
Instead of introducing a new function, we should prepare the code for how it would look in the future, something similar to what we do with httpclient

```

if sys.version_info[0] == 3:	if sys.version_info[0] == 3:
    import http.client as httpclient
elif sys.version_info[0] == 2:	elif sys.version_info[0] == 2:
    import httplib as httpclient

```

Reply:
The problem with that is that we are not importing anything. The API on the `array` data structure changed from `[array].tostring()` to `[array].tobytes()`. Instead of extending the array class and defining our own API depending on the Python version, I found this solution much simpler.

If you have a better idea how to do this, please let me know.

Reply:
if python < 3.9 : tobytes = tostring

Reply:
hold on, this is a class member, 1 sec

Reply:
in what class is tostring() defined?

Reply:
`class array.array`
https://docs.python.org/3/library/array.html#array.array.tobytes

Reply:
ok, since it is a builtin, it won't be possible. I would suggest name the function simply array_to_bytes

Reply:
Yep, this was my conclusion too. Ok, will do the rename.